### DISCUSSION TITLE
Java code to draw molecule from smiles

### DISCUSSION THREAD

**Discussion by tnguy271** [rdkit/rdkit#5622] [Development]
Hi, I'm trying to draw molecule from smiles String in Java. Below is my code, but the returned svg does not depict molecule structure. Thank you for your help.

public static String toImageJava( String smiles){
//
//        RWMol mol = null;
//        if (smiles != null)
//        {
//            mol = RWMol.MolFromSmiles(smiles);
//        }
//        
//        System.out.println(mol.toString());
//        MolDraw2DSVG drawer = new MolDraw2DSVG(400,300);
//
//        drawer.drawMolecule(mol);
//        drawer.finishDrawing();
//        return drawer.getDrawingText();
//
//

**Comment by ptosco** [comment]
@tnguy271 The code you pasted above works for me with the latest RDKit code:
```
import org.RDKit.*;

class drawFromSmiles {
    public static void main(String args[]) {
        System.setProperty("java.library.path", "/scratch/toscopa1/src/rdkit/Code/JavaWrappers/gmwrapper");
        System.loadLibrary("GraphMolWrap");
        String smiles = "O=C(O)C(c1ccc(cc1)CCN4CCC(c2nc3ccccc3n2CCOCC)CC4)(C)C";
        RWMol mol = RWMol.MolFromSmiles(smiles);
        MolDraw2DSVG drawer = new MolDraw2DSVG(400, 300);
        drawer.drawMolecule(mol);
        drawer.finishDrawing();
        System.out.println(drawer.getDrawingText());
    }
}
```
If you are on an older RDKit release you might need to add
```
mol.compute2DCoords();
```
to generate 2D coordinates before drawing the molecule; recent RDKit versions do that automatically in case the molecule has no coordinates.

**Comment by tnguy271** [comment]
Thank you @ptosco. It worked for me after switching to latest version.
